File: jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/UXGraphicsContext.php
<?php
namespace php\gui;
use php\gui\paint\UXColor;
use php\gui\text\UXFont;

/**
 * Class UXGraphicsContext
 * @package php\gui
 */
class UXGraphicsContext
{
    /**
     * @var UXFont
     */
    public $font;

    /**
     * @var float
     */
    public $globalAlpha = 1.0;

    /***
     * SRC_OVER, SRC_ATOP, ADD, MULTIPLY, SCREEN, OVERLAY, DARKEN, LIGHTEN, COLOR_DODGE, COLOR_BURN
     * @var string
     */
    public $globalBlendMode = null;

    /**
     * Fills the path with the current fill paint.
     */
    public function fill()
    {
    }

    /**
     * Fills the given string of text at position x, y
     * with the current fill paint attribute.
     *
     * @param string $text
     * @param double $x
     * @param double $y
     * @param float|int $maxWidth
     */
    public function fillText($text, $x, $y, $maxWidth = 0)
    {
    }

    /**
     * Draws the given string of text at position x, y
     * with the current fill paint attribute.
     *
     * @param string $text
     * @param double $x
     * @param double $y
     * @param float|int $maxWidth
     */
    public function strokeText($text, $x, $y, $maxWidth = 0)
    {
    }

    /**
     * @param $x
     * @param $y
     * @param $w
     * @param $h
     */
    public function clearRect($x, $y, $w, $h)
    {
    }

    /**
     * @param UXImage $image
     * @param $x
     * @param $y
     * @param $w (optional)
     * @param $h (optional)
     * @param $dx (optional)
     * @param $dy (optional)
     * @param $dw (optional)
     * @param $dh (optional)
     */
    public function drawImage(UXImage $image, $x, $y, $w, $h, $dx, $dy, $dw, $dh) {
    }

    /**
     * @param UXColor $color
     */
    public function setFillColor(UXColor $color)
    {
    }
}

File: jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXGraphicsContext.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.scene.canvas.GraphicsContext;
import javafx.scene.effect.BlendMode;
import javafx.scene.image.Image;
import javafx.scene.paint.Color;
import javafx.scene.text.Font;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Abstract;
import php.runtime.annotation.Reflection.Nullable;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.lang.BaseWrapper;
import php.runtime.reflection.ClassEntity;

@Abstract
@Reflection.Name(JavaFXExtension.NS + "UXGraphicsContext")
public class UXGraphicsContext extends BaseWrapper<GraphicsContext> {
    interface WrappedInterface {
        @Property Font font();
        @Property double globalAlpha();
        @Property BlendMode globalBlendMode();

        void fill();
        void fillText(String text, double x, double y);
        void fillText(String text, double x, double y, double maxWidth);

        void strokeText(String text, double x, double y);
        void strokeText(String text, double x, double y, double maxWidth);

        void clearRect(double x, double y, double w, double h);
        void drawImage(Image img, double x, double y);
        void drawImage(Image img, double x, double y, double w, double h);
        void drawImage(Image img, double x, double y, double w, double h, double dx, double dy, double dw, double dh);
    }

    public UXGraphicsContext(Environment env, GraphicsContext wrappedObject) {
        super(env, wrappedObject);
    }

    public UXGraphicsContext(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void setFillColor(@Nullable Color color) {
        getWrappedObject().setFill(color);
    }
}
